app/kuma-dp/pkg/dataplane/accesslogs/streamer.go

```diff
@@ -0,0 +1,147 @@
+//go:build !windows
+
+package accesslogs
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"os"
+	"sync"
+	"syscall"
+
+	"github.com/pkg/errors"
+
+	kumadp "github.com/kumahq/kuma/pkg/config/app/kuma-dp"
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+	"github.com/kumahq/kuma/pkg/xds/envoy"
+)
+
+var logger = core.Log.WithName("access-log-streamer")
+
+var _ component.Component = &accessLogStreamer{}
+
+// accessLogStreamer implements TCP access logging in Kuma.
+// When TCP logging is configured, CP configures Envoy to log access logs to a pipe file.
+// Format of such logs is "address:port;msg"
+// Streamer then reads logs from the pipe and passes it to a TCP destination.
+type accessLogStreamer struct {
+	address string
+
+	sync.RWMutex
+	senders map[string]*logSender
+}
+
+func (s *accessLogStreamer) NeedLeaderElection() bool {
+	return false
+}
+
+func NewAccessLogStreamer(dataplane kumadp.Dataplane) *accessLogStreamer {
+	return &accessLogStreamer{
+		address: envoy.AccessLogSocketName(dataplane.Name, dataplane.Mesh),
+		senders: map[string]*logSender{},
+	}
+}
+
+func (s *accessLogStreamer) Start(stop <-chan struct{}) error {
+	logger.Info("cleaning existing access log pipe", "file", s.address)
+	err := os.Remove(s.address)
```

Comment:
I know it's a behavior we used to have, but is it possible to have a race here? Like Envoy already started writing some logs to the existing pipe and we're deleting it?

Reply:
I did a bit of testing by removing the FIFO in the middle of the process that was running and it recovered. I think I'd leave it for now.